token/transfer-hook-example/src/processor.rs

```diff
@@ -114,14 +116,17 @@ pub fn process_initialize_extra_account_meta_list(
     let signer_seeds = collect_extra_account_metas_signer_seeds(mint_info.key, &bump_seed);
     let length = extra_account_metas.len();
     let account_size = ExtraAccountMetaList::size_of(length)?;
+    let minimum_lamports_for_rent = Rent::get()?.minimum_balance(account_size);
+    let create_instruction = system_instruction::create_account(
+        authority_info.key,
+        extra_account_metas_info.key,
+        minimum_lamports_for_rent,
+        account_size as u64,
+        program_id,
+    );
     invoke_signed(
-        &system_instruction::allocate(extra_account_metas_info.key, account_size as u64),
-        &[extra_account_metas_info.clone()],
-        &[&signer_seeds],
-    )?;
-    invoke_signed(
-        &system_instruction::assign(extra_account_metas_info.key, program_id),
-        &[extra_account_metas_info.clone()],
```

Comment:
This is done deliberately, and not a bug -- since we're defining an interface, we *cannot* expose an instruction that takes a wallet as a signer and writable, since any malicious program implementing the interface can just immediately rug all funds in the user's wallet.

In fact, it's an antipattern to have people pass in their wallet as a signer + writable to any program, especially if the system program is included! They should just transfer the lamports directly to the account separately, that way there's no risk of getting rugged ever.

Was there a reason to change this?

Reply:
Never looked at it this way but it makes sense. I changed this to make the test work which in itself should have been a red flag :))).  Let me change this back.

Reply:
Ok great, thanks!